test-framework/junit5/src/main/java/io/quarkus/test/junit/launcher/CustomLauncherInterceptor.java

```diff
@@ -1,21 +1,29 @@
 package io.quarkus.test.junit.launcher;
 
+import org.junit.platform.launcher.LauncherDiscoveryListener;
+import org.junit.platform.launcher.LauncherDiscoveryRequest;
 import org.junit.platform.launcher.LauncherInterceptor;
 
 import io.quarkus.test.junit.classloading.FacadeClassLoader;
 
-public class CustomLauncherInterceptor implements LauncherInterceptor {
+public class CustomLauncherInterceptor implements LauncherDiscoveryListener, LauncherInterceptor {
 
-    private FacadeClassLoader facadeLoader = null;
-    private ClassLoader origCl = null;
+    private static FacadeClassLoader facadeLoader = null;
+    // Also use a static variable to store a 'first' starting state that we can reset to
+    private static ClassLoader origCl = null;
 
     public CustomLauncherInterceptor() {
     }
 
+    private static boolean isProductionModeTests() {
+        // We're too early for config to be available, so just check the system props
+        return System.getProperty("prod.mode.tests") != null;
```

Comment:
What sets this property?

Reply:
Do you know, I have no idea. Good spot. This is the same code as before, just refactored for clarity. I think I must have found it by inspection of the system properties in native tests, because I can't find it in the source code. There must be a cleaner way, so I'll raise an issue.